cli/tests/ceremony-params/ceremonyParams.test.ts

```diff
@@ -86,75 +105,264 @@ describe("stress tests", function test() {
+        for (let i = 0; i < users.length; i += 1) {
+          const randomVoteOption = BigInt(Math.floor(Math.random() * 126));
+          const randomVoteWeight = BigInt(Math.floor(Math.random() * 10));
+          // eslint-disable-next-line no-await-in-loop
+          await publish({
+            pubkey: users[i].pubKey.serialize(),
+            stateIndex: 1n,
+            voteOptionIndex: randomVoteOption,
+            nonce: 1n,
+            pollId: 0n,
+            newVoteWeight: randomVoteWeight,
+            maciContractAddress: maciAddresses.maciAddress,
+            salt: genRandomSalt(),
+            privateKey: users[i].privKey.serialize(),
+            signer,
+          });
+        }
+      });
+
+      it("should generate zk-SNARK proofs and verify them", async () => {
+        await timeTravel({ seconds: pollDuration, signer });
+        await mergeMessages({ ...mergeMessagesArgs, signer });
+        await mergeSignups({ ...mergeSignupsArgs, signer });
+        await genProofs({ ...genProofsCeremonyArgs, signer });
+        await proveOnChain({ ...proveOnChainArgs, signer });
+        await verify({ ...verifyArgs(), signer });
+      });
     });
+  });
+
+  describe("non quadratic voting", () => {
+    const genProofsArgs: Omit<GenProofsArgs, "signer"> = {
+      outputDir: testProofsDirPath,
+      tallyFile: testTallyFilePath,
+      tallyZkey: ceremonyTallyVotesNonQvZkeyPath,
+      processZkey: ceremonyProcessMessagesNonQvZkeyPath,
+      pollId: 0n,
+      rapidsnark: testRapidsnarkPath,
+      processWitgen: ceremonyProcessMessagesNonQvWitnessPath,
+      processDatFile: ceremonyProcessMessagesNonQvDatPath,
+      tallyWitgen: ceremonyTallyVotesNonQvWitnessPath,
+      tallyDatFile: ceremonyTallyVotesNonQvDatPath,
+      coordinatorPrivKey,
+      processWasm: ceremonyProcessMessagesNonQvWasmPath,
+      tallyWasm: ceremonyTallyVotesNonQvWasmPath,
+      useWasm,
+      useQuadraticVoting: false,
+    };
 
-    it("should signup 1 user", async () => {
-      await signup({ maciAddress: maciAddresses.maciAddress, maciPubKey: users[0].pubKey.serialize(), signer });
+    // before all tests we deploy the vk registry contract and set the verifying keys
+    before(async () => {
+      signer = await getDefaultSigner();
+
+      // we deploy the vk registry contract
+      await deployVkRegistryContract({ signer });
+      // we set the verifying keys
+      await setVerifyingKeys({ ...verifyingKeysNonQvArgs, signer });
     });
 
-    it("should publish 2 messages", async () => {
-      // eslint-disable-next-line @typescript-eslint/prefer-for-of
-      for (let i = 0; i < 2; i += 1) {
-        const randomVoteOption = BigInt(Math.floor(Math.random() * 126));
-        const randomVoteWeight = BigInt(Math.floor(Math.random() * 10));
-        // eslint-disable-next-line no-await-in-loop
+    describe("1 signup, 1 message", () => {
+      after(() => {
+        cleanVanilla();
+      });
+
+      const user = new Keypair();
```

Comment:
why we new a user here instead of using the existing `users` array?

Reply:
no need, removing